### DISCUSSION TITLE
Response model with default value of empty list not showing in docs

### DISCUSSION THREAD

**Discussion by jordantshaw** [fastapi/fastapi#8442] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class User(BaseModel):
    username: str = None

class UserResponse(BaseModel):
    data: List[User] = []
    metadata: Dict
```

### Description

I am having an issue with the docs displaying the response models sub models correctly when an empty list is the default value. I would expect the docs to display the array and the sub-model inside the array, but instead the docs just show an empty array.  If I set the default value to None, it displays as I would expect.  This happens on both redoc and swagger.

Any ideas on a workaround or on how to solve this issue?

This is how the docs appear with when an empty array is used as default value.
![image](https://user-images.githubusercontent.com/33508069/159758477-45b14afa-b575-4753-bca5-3668b24639d7.png)

This is how the docs appear when setting the default value to None (This is what I want the docs to look like).
![image](https://user-images.githubusercontent.com/33508069/159758884-bc498eec-1dfd-458c-a332-6685a1307749.png)

### Operating System

macOS

### FastAPI Version

0.75.0

### Python Version

Python 3.9.10

**Comment by tasercake** [comment]
I agree that this generated example isn't ideal, and I'm not sure if there's a reason behind it or how to change it in the FastAPI source.

A workaround I've used is to override the [`schema_extra`](https://pydantic-docs.helpmanual.io/usage/schema/#schema-customization) field of the response model to include a custom example, which comes with the disadvantage of having to fully specify every field:

```python

from fastapi import FastAPI
from pydantic import BaseModel
from typing import List, Dict, Optional

class Item(BaseModel):
    id: Optional[str] = None

class ItemList(BaseModel):
    items: List[Item] = []
    metadata: Dict

    class Config:
        schema_extra = {
            "example": {
                "items": [Item(id="abc123")],
                "metadata": {"total_items": 1},
            }
        }

app = FastAPI()

@app.get("/")
async def endpoint(items: ItemList):
    pass
```

Which gives me the following in redoc:

<img width="428" alt="image" src="https://user-images.githubusercontent.com/13855549/161224898-1971da97-518a-47a6-8ac9-9fc5ee4a8068.png">

**Comment by teonat** [comment]
Heres an alternative workaround which doesn't require static swagger-documentation. 

Setting the default of the list field to `None` and utilizing a root-validator to set the field to contain an empty list (`[]`) when no data is provided for the field results in swagger showing the model contained in the list properly and for the API-response to return an empty list and not `None`

```py
class User(BaseModel):
    username: str = None

class UserResponse(BaseModel):
    data: List[User] = None
    metadata: Dict

    @root_validator(pre=True)
    def validate_data(cls, data):
        if not data['data']:
            data['data'] = []
        return data
```

**Comment by noamsto** [comment, marked as answer]
It's really shame that it's not being treated on code level.
Anyway another WA is using `Annotated` alongside `default_factory=list`
i.e.:
```python
class Data(Schema):
    relationships: Annotated[list[RelationshipType], Field(default_factory=list)]
```

**Reply by jordantshaw** [reply]
This seems to be the best workaround currently. But still wish there was no need for a workaround.

**Reply by YuriiMotov** [reply]
The problem is that Swagger uses default value as example value.
Not sure FastAPI can solve this since FastAPI doesn't provide example values in neither of these cases.
